snips_nlu/intent_parser/deterministic_intent_parser.py

```diff
@@ -143,8 +157,13 @@ def parse(self, text, intents=None):
                     slots, self.language)
                 parsed_slots = sorted(parsed_slots,
                                       key=lambda s: s[RES_MATCH_RANGE][START])
-                return parsing_result(text, parsed_intent, parsed_slots)
-        return empty_result(text)
+                result = parsing_result(text, parsed_intent, parsed_slots)
+                found_result = True
+                break
+            if found_result:
+                break
+        result = result or empty_result(text)
+        return result
```

Comment:
I think you can rollback this part of the code, as `found_result` and `result` are not needed